zipline/_protocol.pyx

```diff
@@ -20,8 +21,17 @@ import numpy as np
 from cpython cimport bool
 
 from zipline.assets import Asset
+from zipline.zipline_deprecation_warning import ZiplineDeprecationWarning
 
 cdef class BarData:
+    cdef object data_portal
+    cdef object simulation_dt_func
+    cdef object data_frequency
+    cdef object _views
```

Comment:
Does the forward declaration change the generated C?  Should any of these be more explicitly typed, e.g. `cdef dict _views`?